ts/input/tex/ColumnParser.ts

```diff
@@ -131,62 +131,88 @@ export class ColumnParser {
-    //
-    if (state.cspace.length) {
-      const cspace = [...state.cspace];
-      if (cspace.length < calign.length) {
-        cspace.push('1em');
-      }
-      array.arraydef.columnspacing = cspace.slice(1).map(d => d || '1em').join(' ');
+    array.arraydef.columnwidth = cwidth.map(w => w || 'auto').join(' ');
+  }
+
+  /**
+   * @param {ColumnState} state   The current state of the parser
+   * @param {ArrayItem} array     The array stack item to adjust
+   */
+  protected setColumnSpacing(state: ColumnState, array: ArrayItem) {
+    if (!state.cspace.length) return;
+    const cspace = [...state.cspace];
+    if (cspace.length < state.calign.length) {
+      cspace.push('1em');
     }
-    //
-    // Set the column lines and table frame
-    //
-    if (state.clines.length) {
-      const clines = [...state.clines];
-      if (clines[0]) {
-        // @test Enclosed left right, Enclosed left
-        array.frame.push('left');
-        array.dashed = (clines[0] === 'dashed');
-      }
-      if (clines.length > calign.length) {
-        // @test Enclosed left right, Enclosed right
-        array.frame.push('right');
-        clines.pop();
-      } else if (clines.length < calign.length) {
-        clines.push('none');
-      }
+    array.arraydef.columnspacing = cspace.slice(1).map(d => d || '1em').join(' ');
+  }
+
+  /**
+   * @param {ColumnState} state   The current state of the parser
+   * @param {ArrayItem} array     The array stack item to adjust
+   */
+  protected setColumnLines(state: ColumnState, array: ArrayItem) {
+    if (!state.clines.length) return;
+    const clines = [...state.clines];
+    if (clines[0]) {
+      // @test Enclosed left right, Enclosed left
+      array.frame.push(['left', clines[0]]);
+    }
+    if (clines.length > state.calign.length) {
+      // @test Enclosed left right, Enclosed right
+      array.frame.push(['right', clines.pop()]);
+    } else if (clines.length < state.calign.length) {
+      clines.push('none');
+    }
+    if (clines.length > 1) {
       // @test Enclosed left right
       array.arraydef.columnlines =
         clines.slice(1)
-              .map(l => l || 'none')
-              .join(' ');
-    }
-    if (state.cextra[0] || state.cextra[calign.length - 1]) {
-      const cspace = state.cspace;
-      array.arraydef['data-array-padding'] = [
-        cspace[0] || '.5em',
-        cspace[calign.length - 1] || '.5em'
-      ].join(' ');
+        .map(l => l || 'none')
+        .join(' ');
     }
   }
 
+  /**
+   * @param {ColumnState} state   The current state of the parser
+   * @param {ArrayItem} array     The array stack item to adjust
+   */
+  protected setPadding(state: ColumnState, array: ArrayItem) {
+    if (!state.cextra[0] && !state.cextra[state.calign.length - 1]) return;
+    const cspace = state.cspace;
+    array.arraydef['data-array-padding'] = [
+      cspace[0] || '.5em',
+      cspace[state.calign.length - 1] || '.5em'
+    ].join(' ');
```

Comment:
This could be done more efficiently in a template literal:
```suggestion
    array.arraydef['data-array-padding'] = 
       `${cspace[0] || '.5em'} ${cspace[state.calign.length - 1] || '.5em'}`
```

Reply:
Fixed.